_posts/2016-08-02-should-i-use-a-spy-a-double-or-an-instance-double.markdown

```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+title: "Should I use a spy, a double, or an instance_double?"
+date: 2016-08-02 15:03:00
+categories: ["rspec", "ruby"]
+author: "mauro-oto"
+---
+
+When writing tests for services, you may sometimes want to use mock objects
+instead of actual objects, which if you're using ActiveRecord,
+may result in hitting the database and slowing the suite down. The
+latest release of the [rspec-mocks](https://github.com/rspec/rspec-mocks)
+library bundled with RSpec 3 includes at least three different ways to implement
+a mock object.
+
+We'll look at some of the differences in each by looking at some examples. First
+off, the `spy`:
+
+```ruby
+[1] pry(main)> require 'rspec/mocks/standalone'
+=> true
+[2] pry(main)> user_spy = spy(User)
+=> #<Double User>
+[3] pry(main)> spy.whatever_method
+=> #<Double (anonymous)>
+```
+
+<!--more-->
+
+The `spy` accepts any method calls and always returns itself unless specified.
```

Comment:
@etagwerker thanks! I'll add the link to the official doc. 

`expect(spy).to have_received(:whatever_method)` is something the three of them have in common, it's not really a difference, they all implement it. I tried to focus on the fact that if you're already using double, you may as well be using instance_double, as you get more benefits. I'll put that in the conclusion.